### DISCUSSION TITLE
How to fix missing syntax highlighting? (Terraform)

### DISCUSSION THREAD

**Discussion by jsjoeio** [LunarVim/LunarVim#3579] [Q&A]
I want to be able to figure this out on my own next time but here's the problem:
- I open a Terraform file: `main.tf`
- Lunar Vim installs the LSP for me (I think)
- my thinking is that comes with syntax highlighting but it doesn't or I'm missing something?

So...how do I fix missing syntax highlighting for this problem and for the future?

<img width="1668" alt="image" src="https://user-images.githubusercontent.com/3806031/206039488-babd3976-90db-4165-8a6b-686e1cf1c5c9.png">

**Comment by mamachanko** [comment]
> my thinking is [LSP] comes with syntax highlighting but it doesn't or I'm missing something?

Afaik treesitter is responsible for syntax highlighting and not the language server.

**Reply by jsjoeio** [reply]
Oh! TIL. That makes sense then. I don't see Terraform in the support list [here](https://github.com/nvim-treesitter/nvim-treesitter#supported-languages).

**Reply by jsjoeio** [reply]
Huh, but here in the `terraform-ls` docs they mention [syntax-highlighting](https://github.com/hashicorp/terraform-ls/blob/main/docs/syntax-highlighting.md).

**Reply by LostInTheLogs** [reply]
Neovim doesn't support syntax highlighting from lsp, there was a or merged recently for semantic highlighting, idk if that's the same thing or not though

**Reply by jsjoeio** [reply]
Ahh got it! I don't know what I did differently but reinstalling LunarVim and now it works lol
